scripts/check_catalog_schema.py

```diff
@@ -15,15 +14,33 @@
 def main(directory: Path) -> int:
     code = 0
     for org_path in get_schema_paths(directory):
-        report = validate(org_path)
-        if not report.valid:
-            print(format_error_message(f"{org_path} schema is not valid"))
+        try:
+            schema = Schema(org_path)
+            report = schema.validate()
+            if not report.valid:
+                print(format_error_message(f"Error: {org_path} schema is not valid"))
+                code = 1
+                errors = report.flatten(["code", "message"])
+                print(tabulate(errors, headers=["code", "message"]))
+                continue
+
+            missing_fields = get_missing_fields(schema.field_names)
+
+            if len(missing_fields) > 0:
+                missing_fields_string = ",".join(missing_fields)
+                print(
+                    format_error_message(
+                        f"Error: {missing_fields_string} fields are missing in the schema in {org_path}"
+                    )
+                )
+                code = 1
+        except:
```

Comment:
Il serait préférable de définir quelle(s) exception(s) on s'attend à recevoir

En effet, si tu as une erreur de programmation inattendue (bug), actuellement on passe dans cette branche et on affiche seulement "An error has occurred" au lieu de laisser l'erreur remontée et être affichée dans un traceback détaillé.

D'ailleurs si tu as le linter flake8 activé, il devrait te dire "Do not use bare except"

Pour cette même raison de débogage et d'autres (lisibilité), il est préférable d'encadrer la plus petite portion de code pouvant lever une exception

Enfin, à defaut d'afficher un traceback complet, on devrait au moins afficher l'erreur dans le message :


```python
except (FirstException, SecondException) as exc:  # Lesquelles ? À définir
    print(format_error_message(f"Error: {exc}"))
```